src/integrationTest/java/com/mongodb/hibernate/EmbeddableIntegrationTests.java

```diff
@@ -0,0 +1,272 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate;
+
+import static com.mongodb.hibernate.MongoTestAssertions.assertEquals;
+import static com.mongodb.hibernate.internal.MongoConstants.ID_FIELD_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.mongodb.client.MongoCollection;
+import com.mongodb.hibernate.junit.InjectMongoCollection;
+import com.mongodb.hibernate.junit.MongoExtension;
+import jakarta.persistence.AttributeOverride;
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.hibernate.annotations.Struct;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.SessionFactoryScopeAware;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@SessionFactory(exportSchema = false)
+@DomainModel(
+        annotatedClasses = {
+            EmbeddableIntegrationTests.ItemWithFlattenedValues.class,
+            EmbeddableIntegrationTests.ItemWithNestedValues.class,
+            EmbeddableIntegrationTests.Unsupported.ItemWithNestedValueHavingNonInsertable.class,
+            EmbeddableIntegrationTests.Unsupported.ItemWithNestedValueHavingAllNonInsertable.class,
+            EmbeddableIntegrationTests.Unsupported.ItemWithNestedValueHavingNonUpdatable.class
+        })
+@ExtendWith(MongoExtension.class)
+class EmbeddableIntegrationTests implements SessionFactoryScopeAware {
+    @InjectMongoCollection("items")
+    private static MongoCollection<BsonDocument> mongoCollection;
+
+    private SessionFactoryScope sessionFactoryScope;
+
+    @Test
+    void flattenedValues() {
+        var item = new ItemWithFlattenedValues();
+        item.id = new EmbeddableValue();
+        item.id.a = 1;
+        item.flattened1 = new EmbeddableValue();
+        item.flattened1.a = 2;
+        item.flattened2 = new EmbeddablePairValue1(3, new EmbeddablePairValue2(4, 5));
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
```

Comment:
Hibernate ORM 7.0 deprecated more Hibernate-specific methods on `SessionFactory` in favour of those from Jakarta `EntityManager`. We should start using them now, instead of producing more code that will have to be changed once we upgrade to Hibernate ORM 7.0.

Reply:
replacing `get` with `find` method is fine, but given we are still using v6.6, it is not urgent to embrace Jakarta for now; when we switch to v7, it seems pretty straightforward to refactor for I don't think we use Hibernate-specific methods (which have no Jakarta counterparts) a lot.